### PR TITLE
bump ffmpeg to 7 and enable arm64 osx 

### PR DESCRIPTION
**Pull request by botanicvelious**

### CODE REVIEW

**Comment by botanicvelious** [comment]
Required to enable arm64 osx builds in https://github.com/OpenEnroth/OpenEnroth/pull/1538

File: .github/workflows/all.yml [hunk 1]
```diff
@@ -145,6 +145,8 @@ jobs:
           ./scripts/print_info.sh
 
       - name: 'Publish release'
+        # Dont try to publish for PR's
+        if: github.event_name != 'pull_request'
```
Comment (captainurist) [inline .github/workflows/all.yml:149]: Why? This was actually handy for testing.

File: scripts/build_all.sh [hunk 2]
```diff
@@ -260,10 +260,6 @@ function ffmpeg_install() {
         "--enable-avutil" \
         "--enable-swresample" \
         "--enable-swscale" \
-        "--enable-dct" \
-        "--enable-mdct" \
-        "--enable-rdft" \
-        "--enable-fft" \
```
Comment (captainurist) [inline scripts/build_all.sh:266]: Why?

File: .github/workflows/all.yml (hunk 1 above)
Comment (botanicvelious) [reply .github/workflows/all.yml:149]: doesnt work for anyone but you I assume ill change it to just not fail on it

File: scripts/build_all.sh (hunk 2 above)
Comment (botanicvelious) [reply scripts/build_all.sh:266]: ffmpeg 7 removed those options :)

**Pull request merged**